argh_derive/src/lib.rs

```diff
@@ -411,14 +426,29 @@ fn impl_from_args_struct_redact_arg_values<'a>(
     let redact_subcommands = if let Some(subcommand) = subcommand {
         let name = subcommand.name;
         let ty = subcommand.ty_without_wrapper;
-        quote_spanned! { impl_span =>
-            Some(argh::ParseStructSubCommand {
-                subcommands: <#ty as argh::SubCommands>::COMMANDS,
-                parse_func: &mut |__command, __remaining_args| {
-                    #name = Some(<#ty as argh::FromArgs>::redact_arg_values(__command, __remaining_args)?);
-                    Ok(())
-                },
-            })
+
+        if let Some(ref default) = subcommand.attrs.default {
+            quote_spanned! { impl_span =>
+                Some(argh::ParseStructSubCommand {
+                    subcommands: <#ty as argh::SubCommands>::COMMANDS,
+                    parse_func: &mut |__command, __remaining_args| {
+                        #name = Some(<#ty as argh::FromArgs>::redact_arg_values(__command, __remaining_args)?);
+                        Ok(())
+                    },
+                    default: Some( #default ),
+                })
+            }
+        } else {
+            quote_spanned! { impl_span =>
+                Some(argh::ParseStructSubCommand {
+                    subcommands: <#ty as argh::SubCommands>::COMMANDS,
+                    parse_func: &mut |__command, __remaining_args| {
+                        #name = Some(<#ty as argh::FromArgs>::redact_arg_values(__command, __remaining_args)?);
+                        Ok(())
+                    },
+                    default: None,
+                })
+            }
```

Comment:
I think we can simplify this and remove some code duplication by doing:

```suggestion
        let default_opt = if let Some(ref default) = subcommand.attrs.default {
            quote! { Some ( #default ) }
        } else {
            quote! { None }
        };
        quote_spanned! { impl_span =>
            Some(argh::ParseStructSubCommand {
                subcommands: <#ty as argh::SubCommands>::COMMANDS,
                parse_func: &mut |__command, __remaining_args| {
                    #name = Some(<#ty as argh::FromArgs>::redact_arg_values(__command, __remaining_args)?);
                    Ok(())
                },
                default: #default_opt,
            })
        }
```

Reply:
I tried to do this earlier but could not figure out the span library. Thanks.